var/spack/repos/builtin/packages/py-skl2onnx/package.py

```diff
@@ -21,3 +22,4 @@ class PySkl2onnx(PythonPackage):
     depends_on("py-onnx@1.2.1:", type=("build", "run"))
     depends_on("py-scikit-learn@0.19:", type=("build", "run"))
     depends_on("py-onnxconverter-common@1.7.0:", type=("build", "run"))
+    depends_on("py-packaging", when="@1.12:", type=("build", "run"))
```

Comment:
Might be useful to add a comment saying that although this dep is undocumented, it's imported at run-time.